src/mixins/README.md

```diff
@@ -0,0 +1,128 @@
+# Mixins
+
+This directory contains source files for the mixins exported by this extension.
+
+## Overview
+
+Sometimes it's helpful to write partial classes and then combining them together to build more powerful classes. This pattern is called Mixins (mixing in partial classes) and is supported by LoopBack 4. 
+
+LoopBack 4 supports mixins at an `Application` level. Your partial class can then be mixed into the `Application` class. A mixin class can modify or override existing methods of the class or add new ones! It is also possible to mixin multiple classes together as needed.
+
+### High level example
+```ts
+const MixedClass = MyMixinClass(Application);
+const MultipleMixedClassesClass = MyMixinClass(MyMixinClass2(Application));
+```
+
+
+## Getting Started
+
+For hello-extensions we write a simple Mixin that allows the `Application` class to bind a `Logger` class from ApplicationOptions, Components, or `.logger()` method that is mixed in. `Logger` instances are bound to the key `loggers.${Logger.name}`. Once a Logger has been bound, the user can retrieve it by using [Dependency Injection](http://loopback.io/doc/en/lb4/Dependency-injection.html) and the key for the `Logger`.
+
+### What is a Logger?
+> A Logger class is provides a mechanism for logging messages of varying priority by providing an implementation for `Logger.info()` & `Logger.error()`. An example of a Logger is `console` which has `console.log()` and `console.error()`. 
+
+#### An example Logger
+```ts
+class ColorLogger implements Logger {
+  log(...args: any[]) {
+    console.log('log  :', ...args);
+  }
+
+  error(...args: any[]) {
+    const data = args.join(' ');
+    // log in red color
+    console.log('\x1b[31m error: ' + data + '\x1b[0m');
+  }
+}
+```
+
+## LoggerMixin
+A complete & functional implementation can be found in `logger.mixin.ts`. *Here are some key things to keep in mind when writing your own Mixin*.
+
+### constructor()
+A Mixin constructor must take an array of any type as it's argument. This would represent `ApplicationOptions` for our base class `Application` as well as any properties we would like for our Mixin. 
+
+It is also important for the constructor to call `super(args)` so `Application` continues to work as expected.
+```ts
+constructor(...args: any[]) {
+  super(args);
+}```
+
+### Binding via `ApplicationOptions`
+As mentioned earlier, since our `args` represents `ApplicationOptions`, we can make it possible for users to pass in their `Logger` implementations in a `loggers` array on `ApplicationOptions`. We can then read the array and automatically bind these for the user. 
+
+#### Example user experience
+```ts
+new LoggerMixin(Application)({
```

Comment:
Sorry, this is not how most people are going to use a mixin. Please save the new app to a variable (or create a new class extending the mixed-in app) and then call `new` on this new variable/class.